checkBalanced.js

```diff
@@ -0,0 +1,96 @@
+class TreeNode {
+    constructor(data) {
+        this.data = data
+        this.left = null
+        this.right = null
+    }
+}
+
+class Tree {
+    constructor() {
+        this.head = null
+    }
+
+    add(data) {
+        const newNode = new TreeNode(data)
+        if (!this.head) {
+            this.head = newNode
+            return
+        }
+        this.addNode(this.head, newNode)
+    }
+
+    addNode(node, newNode) {
+        if (newNode.data < node.data) {
+            if (!node.left) {
+                node.left = newNode
+            } else {
+                this.addNode(node.left, newNode)
+            }
+        } else {
+            if (!node.right) {
+                node.right = newNode
+            } else {
+                this.addNode(node.right, newNode)
+            }
+        }
```

Comment:
Since you only have an `if-else` in your `else` case, this can be simplified to an `else if` and an `else`.
```suggestion
        } else if (!node.right) {
            node.right = newNode
        } else {
            this.addNode(node.right, newNode)
        }
```